musit-models/src/main/scala/no/uio/musit/models/GroupModule.scala

```diff
@@ -0,0 +1,36 @@
+package no.uio.musit.models
+
+import play.api.libs.json._
+
+sealed trait GroupModule {
+  val id: Int
+}
+
+case object StorageFacility extends GroupModule { val id = 1 }
+case object Loan extends GroupModule { val id = 2 }
+case object Analysis extends GroupModule { val id = 3 }
+
+object GroupModule {
+
+  implicit val reads: Reads[GroupModule] = Reads { jsv =>
+    jsv.validate[Int] match {
+      case JsSuccess(id, path) =>
+        fromInt(id).map(JsSuccess(_)).getOrElse(JsError(path, s"Unknown Module ID $id"))
+
+      case err: JsError =>
+        err
+    }
+  }
+
+  implicit val writes: Writes[GroupModule] = Writes{ m => JsNumber(m.id)}
+
+  @throws(classOf[NoSuchElementException])
+  def unsafeFromInt(id: Int): GroupModule = fromInt(id).get
```

Comment:
When I program in Java, I tend to avoid catching null pointer exceptions that should never occur, because, they should never occur. Our discussion above, centers around the fact that there is not a single endpoint that allows you to post an invalid moduleId. If its invalid, its rejected. Therefore I still think its unnecessary with an unsafeFromInt, in general, due to the fact that what we are actually guarding against is not actually invalid ids, because thats impossible, but we are guarding against debugging a hard to understand stacktrace that tells us that a value fell through a case match.